File: TopInterview150/Hashmap/TwoSum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        vector<int> output;
        for (int i = 0; i < nums.size(); i++)
        {
            if (map.find(target-nums[i]) != map.end())
            {
                output.push_back(map[target-nums[i]]);
                output.push_back(i);
                return output;
            }
            map[nums[i]] = i;
        }
        return output;
    }
};

File: TopInterview150/Hashmap/TwoSum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for i in range(len(nums)):
            if target-nums[i] in map:
                return [i, map[target-nums[i]]]
            map[nums[i]] = i
        return [0, 0]